joern-cli/frontends/php2cpg/src/test/scala/io/joern/php2cpg/querying/TypeDeclTests.scala

```diff
@@ -507,4 +529,128 @@ class TypeDeclTests extends PhpCode2CpgFixture {
       case xs => fail(s"expected localNode and body, got ${xs.code.mkString("[", ",", "]")}")
     }
   }
+
+  "Singleton type decl for a normal class" should {
+    val cpg = code("""<?php
+        |class Foo {
+        |  public static $foo = 'foo';
+        |  public $baz = 'baz';
+        |  const BAZZ = 'bazz';
+        |  public static function bar() {}
+        |}
+        |""".stripMargin)
+
+    "create a singleton type decl" in {
+      inside(cpg.typeDecl.name("Foo.<class>").l) {
+        case fooTypeDecl :: Nil =>
+          fooTypeDecl.modifier.modifierType.l shouldBe List(ModifierTypes.STATIC)
```

Comment:
I see no reason to put the STATIC modifier on the meta class type decl. That is not indicated by the php code and not needed for our modeling.